lib/core/src/Cardano/Wallet/DB.hs

```diff
@@ -358,6 +359,8 @@ newtype ErrNoSuchWallet
     = ErrNoSuchWallet WalletId -- Wallet is gone or doesn't exist yet
     deriving (Eq, Show)
 
+instance Exception ErrNoSuchWallet
```

Comment:
To make review possible, please only rearrange modules during this Pull Request — additions and deletions of stuff go into a separate Pull Request. 🤓

As it turns out, this `instance Exception ErrNoSuchWallet` is surprisingly controversial. 😱

Reply:
how SomeException is supposed to be used for `loadS :: Either SomeException a`? In this case, I think asking for a non-stored wallet id has already an error, namely `ErrNoSuchWallet`. I feel introducing a new  duplicate error is not what we want. What is actually controversial here?

Reply:
> What is actually controversial here?

Long story. In the wallet, we use a lot `ExceptT e IO`. This is an example of a "checked exception" where the exception type `e` is visible on the type level. But in order for this to make sense, `e` should _not_ be an instance of the `Exception` class — to guarantee that `e` cannot be thrown by the `IO` portion. That's the main reason.

When designing `loadS :: Either SomeException a`, I intentionally did not use checked exceptions in order to simplify the types (one less type parameter). But this is somewhat at odds with the checked exceptions style of `ExceptT`.

(My personal preference, and common recommendation, is to not use `ExceptT`. I feel that the time spent on converting different error types into each other could be spent on making the code more correct instead.)